presto-spi/src/main/java/com/facebook/presto/spi/type/SqlTime.java

```diff
@@ -19,30 +19,47 @@
 import java.time.ZoneId;
 import java.time.format.DateTimeFormatter;
 import java.util.Objects;
+import java.util.Optional;
+
+import static com.facebook.presto.spi.type.TimeZoneKey.UTC_KEY;
 
 public final class SqlTime
 {
     private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("HH:mm:ss.SSS");
 
     private final long millisUtc;
-    private final TimeZoneKey sessionTimeZoneKey;
+    private final Optional<TimeZoneKey> sessionTimeZoneKey;
+
+    public SqlTime(long millisUtc)
+    {
+        this.millisUtc = millisUtc;
+        this.sessionTimeZoneKey = Optional.empty();
+    }
 
+    @Deprecated
     public SqlTime(long millisUtc, TimeZoneKey sessionTimeZoneKey)
     {
         this.millisUtc = millisUtc;
-        this.sessionTimeZoneKey = sessionTimeZoneKey;
+        this.sessionTimeZoneKey = Optional.of(sessionTimeZoneKey);
     }
 
     public long getMillisUtc()
     {
         return millisUtc;
     }
 
-    public TimeZoneKey getSessionTimeZoneKey()
+    @Deprecated
+    public Optional<TimeZoneKey> getSessionTimeZoneKey()
     {
         return sessionTimeZoneKey;
     }
 
+    @Deprecated
+    private boolean isDeprecatedTimestamp()
```

Comment:
I would keep it aligned across the code, it will be easier to remove all those functions and variables when we decide to drop backward compatibility.